hydroflow_lang/src/graph/ops/halfjoin.rs

```diff
@@ -0,0 +1,143 @@
+use super::{
+    parse_single_lifetime, OperatorConstraints, OperatorWriteOutput, Persistence, WriteContextArgs,
+    WriteIteratorArgs, RANGE_1,
+};
+
+use quote::quote_spanned;
+use syn::parse_quote;
+
+/// > 2 input streams of type <(K, V1)> and <(K, V2)>, 1 output stream of type <(K, (V1, V2))>
+///
+/// Forms the "streaming half-join" of the tuples in the input streams by their first (key)
```

Comment:
This should not be an op in the surface syntax, should it? 

Either way, the documentation here is not on target.

Reply:
If this is the canonical relational Join operator, don't call the file `halfjoin.rs`?